### ISSUE TITLE
Back button takes back to registration page (and semi-unregistered state)

### ISSUE THREAD

**Issue by sudhirkhanger** [signalapp/Signal-Android#4928]
I setup Signal on Android and it worked fine. Until I pressed back button and I am taken back to registration page. Now when I open the app I get a message to `Enable Signal messages and calls`. I am unregistered just because I pressed back button.

**Comment by mwaclawek** [comment]
Same problem here!

In my case, I'm taken back to the registration page (phone number verification) whenever I press the back button on the main screen (list of chats).

But when I close the app with the home button and open it again via its icon on the home screen, I'm in the list of chats again, as expected. Nevertheless, pressing the back button there leads to the problem described previously, again.

What I remember is that, initially, phone number verification was not successful on this device. I had to resort to the alternative method (automatic phone call). On all the other devices where phone number verification was successful at the first attempt, the back button issue does not happen.

**Comment by FeuRenard** [comment]
We could try [`FLAG_ACTIVITY_CLEAR_TOP`](https://developer.android.com/reference/android/content/Intent.html#FLAG_ACTIVITY_CLEAR_TOP) in [`RegistrationProgressActivity`](https://github.com/WhisperSystems/Signal-Android/blob/master/src/org/thoughtcrime/securesms/RegistrationProgressActivity.java#L336).

**Comment by FeuRenard** [comment]
I tried this with a virtual device Nexus 5X on Android 6.0, Signal 3.24.1 and cannot reproduce this behaviour. I tried re-registering and register directly after reinstall (without seeing the conversation list, Signal jumps directly to the registration). I had to do voice verification as a virtual device cannot receive SMS. After successful registration the conversation list appears and pressing back leads me to the home screen.
